fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/ProgressiveLoanScheduleGenerator.java

```diff
@@ -103,11 +104,15 @@ public LoanScheduleModel generate(final MathContext mc, final LoanApplicationTer
 
         prepareDisbursementsOnLoanApplicationTerms(loanApplicationTerms);
 
+        final ArrayList<DisbursementData> disbursementDataList = new ArrayList<>(loanApplicationTerms.getDisbursementDatas());
+        disbursementDataList.sort(Comparator.comparing(DisbursementData::disbursementDate));
```

Comment:
I'm having the same question here as for PR https://github.com/apache/fineract/pull/4048

Reply:
This branch was prepared to check some test cases on another feature branch.
Now rebased and up to date. I removed the unnecessary commits from this branch.